### DISCUSSION TITLE
joining nodes to the the rke2 server node

### DISCUSSION THREAD

**Discussion by KarimaFathi** [rancher/rke2#4513] [Q&A] [answered]
After configuring the Node server and confirming that it is in the running state, I attempted to join the other nodes using the Node server token and the URL, following the documentation provided at https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-cluster-setup/rke2-for-rancher. However, when trying to start the node, I encountered the following status:

```
--
-- The result is failed.
Jul 26 20:50:39 HA-Rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 26 20:50:39 HA-Rancher-2 systemd[1]: rke2-server.service failed.
Jul 26 20:50:44 HA-Rancher-2 systemd[1]: rke2-server.service holdoff time over, scheduling restart.
Jul 26 20:50:44 HA-Rancher-2 systemd[1]: Stopped Rancher Kubernetes Engine v2 (server).
-- Subject: Unit rke2-server.service has finished shutting down
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit rke2-server.service has finished shutting down.
Jul 26 20:50:44 HA-Rancher-2 systemd[1]: Starting Rancher Kubernetes Engine v2 (server)...
-- Subject: Unit rke2-server.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit rke2-server.service has begun starting up.
Jul 26 20:50:44 HA-Rancher-2 sh[9427]: + /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
Jul 26 20:50:44 HA-Rancher-2 sh[9427]: Failed to get unit file state for nm-cloud-setup.service: No such file or directory
Jul 26 20:50:45 HA-Rancher-2 rke2[9436]: time="2023-07-26T20:50:45Z" level=warning msg="not running in CIS mode"
Jul 26 20:50:45 HA-Rancher-2 rke2[9436]: time="2023-07-26T20:50:45Z" level=info msg="Starting rke2 v1.24.15+rke2r1 (8cf3a75d5
Jul 26 20:50:45 HA-Rancher-2 rke2[9436]: time="2023-07-26T20:50:45Z" level=fatal msg="starting kubernetes: preparing server:
Jul 26 20:50:45 HA-Rancher-2 systemd[1]: rke2-server.service: main process exited, code=exited, status=1/FAILURE
Jul 26 20:50:45 HA-Rancher-2 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
-- Subject: Unit rke2-server.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit rke2-server.service has failed.
--
-- The result is failed.
Jul 26 20:50:45 HA-Rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 26 20:50:45 HA-Rancher-2 systemd[1]: rke2-server.service failed.
```

**Comment by brandond** [comment]
> `Jul 26 20:50:45 HA-Rancher-2 rke2[9436]: time="2023-07-26T20:50:45Z" level=fatal msg="starting kubernetes: preparing server:`

You've truncated the log lines so the error message is not visible. Please attach the **complete** output of `journalctl -u rke2-server --no-pager > rke2-server.log`

**Comment by KarimaFathi** [comment]
oh yes, I can see that the problem is related to the dns url: 

```
Jul 26 20:59:36 HA-Rancher-2 rke2[11777]: time="2023-07-26T20:59:36Z" level=fatal msg="starting kubernetes: preparing server: failed to get CA certs: Get \"https://rancher.inwi.priv:9345/cacerts\": dial tcp 172.20.10.210:9345: connect: no route to host"
Jul 26 20:59:36 HA-Rancher-2 systemd[1]: rke2-server.service: main process exited, code=exited, status=1/FAILURE
Jul 26 20:59:36 HA-Rancher-2 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
Jul 26 20:59:36 HA-Rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 26 20:59:36 HA-Rancher-2 systemd[1]: rke2-server.service failed.
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: rke2-server.service holdoff time over, scheduling restart.
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: Stopped Rancher Kubernetes Engine v2 (server).
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: Starting Rancher Kubernetes Engine v2 (server)...
Jul 26 20:59:41 HA-Rancher-2 sh[11792]: + /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
Jul 26 20:59:41 HA-Rancher-2 sh[11792]: Failed to get unit file state for nm-cloud-setup.service: No such file or directory
Jul 26 20:59:41 HA-Rancher-2 rke2[11801]: time="2023-07-26T20:59:41Z" level=warning msg="not running in CIS mode"
Jul 26 20:59:41 HA-Rancher-2 rke2[11801]: time="2023-07-26T20:59:41Z" level=info msg="Starting rke2 v1.24.15+rke2r1 (8cf3a75d5ccd6e2aa0a99cdf869426f1decd970d)"
Jul 26 20:59:41 HA-Rancher-2 rke2[11801]: time="2023-07-26T20:59:41Z" level=fatal msg="starting kubernetes: preparing server: failed to get CA certs: Get \"https://rancher.inwi.priv:9345/cacerts\": dial tcp 172.20.10.210:9345: connect: no route to host"
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: rke2-server.service: main process exited, code=exited, status=1/FAILURE
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 26 20:59:41 HA-Rancher-2 systemd[1]: rke2-server.service failed.
```
but I don't know why this problem exists because both the load balancer and the dns servers are running and tha machine itself can resolve  the domain name : 

<img width="322" alt="image" src="https://github.com/rancher/rke2/assets/135225708/7acc4177-9be3-4205-b8b1-6c5a80fa6429">

**Reply by brandond** [reply]
It doesn't say that it can't resolve it, it says that it doesn't have a route to it:
`dial tcp 172.20.10.210:9345: connect: no route to host`

Are you sure the hosts and load-balancers are on attached network segments, with the proper routes in place between them? Are the correct ports open?

**Reply by KarimaFathi** [reply]
the load balancer and the dns server exist in the same machine which means the LB has the same IP and yes as you can see :  

```
# Nginx configuration generated by Ansible

load_module '/usr/lib64/nginx/modules/ngx_stream_module.so';
worker_processes 4;
worker_rlimit_nofile 40000;

events {
    worker_connections 8192;
}

stream {
       upstream rancher_servers_http {
        least_conn;
                server 172.20.10.11:80 max_fails=3 fail_timeout=5s;
                server 172.20.10.13:80 max_fails=3 fail_timeout=5s;
                server 172.20.10.14:80 max_fails=3 fail_timeout=5s;
            }
      upstream rancher_servers_https {
        least_conn;
                server 172.20.10.11:443 max_fails=3 fail_timeout=5s;
                server 172.20.10.13:443 max_fails=3 fail_timeout=5s;
                server 172.20.10.14:443 max_fails=3 fail_timeout=5s;
            }
      upstream RKE2_supervisor_backend {
        least_conn;
                server 172.20.10.11:9345 max_fails=3 fail_timeout=5s;
                server 172.20.10.13:9345 max_fails=3 fail_timeout=5s;
                server 172.20.10.14:9345 max_fails=3 fail_timeout=5s;
            }
      upstream kubernetes_api_backend {
        least_conn;
                server 172.20.10.11:6443 max_fails=3 fail_timeout=5s;
                server 172.20.10.13:6443 max_fails=3 fail_timeout=5s;
                server 172.20.10.14:6443 max_fails=3 fail_timeout=5s;
            }


        server {
        listen 80;
        proxy_pass rancher_servers_http;
    }
        server {
        listen 443;
        proxy_pass rancher_servers_https;
    }
        server {
        listen 9345;
        proxy_pass RKE2_supervisor_backend;
    }
        server {
        listen 6443;
        proxy_pass kubernetes_api_backend;
    }

}
```
<img width="416" alt="image" src="https://github.com/rancher/rke2/assets/135225708/025ea4d5-99ee-4f95-be3b-5c024c9cf0a8">
<img width="428" alt="image" src="https://github.com/rancher/rke2/assets/135225708/77b114b8-d26e-44a9-a087-287d5e3ef032">
<img width="365" alt="image" src="https://github.com/rancher/rke2/assets/135225708/d27f98dd-5336-4837-aac6-5efcb15718e8">

**Reply by brandond** [reply]
Can you `curl -vks https://rancher.inwi.priv:9345/cacerts` from the command-line on your nodes?

**Reply by KarimaFathi** [reply]
`[root@HA-Rancher-0 ~]# curl -vks https://rancher.inwi.priv:9345/cacerts
* About to connect() to proxy 10.137.40.42 port 3128 (#0)
*   Trying 10.137.40.42...
* Connected to 10.137.40.42 (10.137.40.42) port 3128 (#0)
* Establish HTTP proxy tunnel to rancher.inwi.priv:9345
> CONNECT rancher.inwi.priv:9345 HTTP/1.1
> Host: rancher.inwi.priv:9345
> User-Agent: curl/7.29.0
> Proxy-Connection: Keep-Alive
>
< HTTP/1.1 403 Forbidden
< Server: squid/3.5.20
< Mime-Version: 1.0
< Date: Wed, 26 Jul 2023 20:50:09 GMT
< Content-Type: text/html;charset=utf-8
< Content-Length: 3504
< X-Squid-Error: ERR_ACCESS_DENIED 0
< Vary: Accept-Language
< Content-Language: en
< X-Cache: MISS from squid
< X-Cache-Lookup: NONE from squid:3128
< Via: 1.1 squid (squid/3.5.20)
< Connection: keep-alive
<
* Received HTTP code 403 from proxy after CONNECT
* Connection #0 to host 10.137.40.42 left intact
`

**Reply by brandond** [reply]
it appears that you're going through a proxy server. Did you need to add `*.inwi.priv` to the NO_PROXY variable? See https://docs.rke2.io/advanced#configuring-an-http-proxy

**Reply by brandond** [reply]
Yes, I suspect you'll need to tinker with both the proxy configuration, and your nginx setup, until you figure out what's preventing you from accessing RKE2 through the load-balancer.

**Reply by KarimaFathi** [reply]
yes, there was a problem with the proxy conf, but I still get the no route to host error 
<img width="295" alt="image" src="https://github.com/rancher/rke2/assets/135225708/fed42426-6285-4b8d-a4f2-2fb690685198">

**Reply by brandond** [reply]
Until you can curl that successfully, RKE2 is not going to work either. Have you confirmed that the correct ports are open between the servers and nginx, and nginx and the servers? Do the nginx access or error logs contain any clues?

**Reply by KarimaFathi** [reply]
The curl is now working successfully. Thanks for your helpful ideas, but the problem is that I couldn't join the node to the server. After investigating, I found that the server logs are showing:

`Jul 26 23:29:22 HA-Rancher-0 rke2[2820]: time="2023-07-26T23:29:22Z" level=error msg="Failed to get recorded learner progress from etcd: context deadline exceeded" `

and the node  that I want to join is showing these logs : 

`Jul 26 23:16:17 ha-rancher-2 rke2[20905]: {"level":"warn","ts":"2023-07-26T23:16:17.482Z","logger":"etcd-client","caller":"v3@v3.5.4-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00079aa80/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:2379: connect: connection refused\""}
Jul 26 23:16:17 ha-rancher-2 rke2[20905]: time="2023-07-26T23:16:17Z" level=error msg="Failed to check local etcd status for learner management: context deadline exceeded"
Jul 26 23:16:17 ha-rancher-2 rke2[20905]: time="2023-07-26T23:16:17Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
`

**Reply by brandond** [reply]
Check the etcd pod logs under /var/log/pods/ to see why etcd isn't responding.

**Reply by KarimaFathi** [reply]
Hi, I tried again this morning to join the node, but unfortunately, I am step back from where I was yesterday. The logs from the agent node are stating:
` Jul 27 13:10:22 ha-rancher-2 rke2[31465]: time="2023-07-27T13:10:22Z" level=fatal msg="starting kubernetes: preparing server: failed to get CA certs: https://rancher.inwi.priv:9345/cacerts: 503 Service Unavailable"
Jul 27 13:10:22 ha-rancher-2 systemd[1]: rke2-server.service: main process exited, code=exited, status=1/FAILURE
Jul 27 13:10:22 ha-rancher-2 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
Jul 27 13:10:22 ha-rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 27 13:10:22 ha-rancher-2 systemd[1]: rke2-server.service failed.
Jul 27 13:10:28 ha-rancher-2 systemd[1]: rke2-server.service holdoff time over, scheduling restart.
Jul 27 13:10:28 ha-rancher-2 systemd[1]: Stopped Rancher Kubernetes Engine v2 (server).
Jul 27 13:10:28 ha-rancher-2 systemd[1]: Starting Rancher Kubernetes Engine v2 (server)...
Jul 27 13:10:28 ha-rancher-2 sh[31479]: + /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
Jul 27 13:10:28 ha-rancher-2 sh[31479]: Failed to get unit file state for nm-cloud-setup.service: No such file or directory
Jul 27 13:10:28 ha-rancher-2 rke2[31485]: time="2023-07-27T13:10:28Z" level=warning msg="not running in CIS mode"
Jul 27 13:10:28 ha-rancher-2 rke2[31485]: time="2023-07-27T13:10:28Z" level=info msg="Starting rke2 v1.24.15+rke2r1 (8cf3a75d5ccd6e2aa0a99cdf869426f1decd970d)"
Jul 27 13:10:28 ha-rancher-2 rke2[31485]: time="2023-07-27T13:10:28Z" level=info msg="Managed etcd cluster not yet initialized"
Jul 27 13:10:28 ha-rancher-2 rke2[31485]: time="2023-07-27T13:10:28Z" level=fatal msg="starting kubernetes: preparing server: failed to validate server configuration: CA cert validation failed: https://rancher.inwi.priv:9345/cacerts: 503 Service Unavailable"
Jul 27 13:10:28 ha-rancher-2 systemd[1]: rke2-server.service: main process exited, code=exited, status=1/FAILURE
Jul 27 13:10:28 ha-rancher-2 systemd[1]: Failed to start Rancher Kubernetes Engine v2 (server).
Jul 27 13:10:28 ha-rancher-2 systemd[1]: Unit rke2-server.service entered failed state.
Jul 27 13:10:28 ha-rancher-2 systemd[1]: rke2-server.service failed.`

I attempted to investigate the problem, but what I find strange is that the 'curl' command works intermittently, which is confusing.
 
<img width="242" alt="image" src="https://github.com/rancher/rke2/assets/135225708/ca318602-884e-4101-b470-27c5be0fcbaa">


With the "netstat" command form the LB vm, I can observe that the connection state remains in "TIME_WAIT" for over 2 minutes instead of transitioning to "ESTABLISHED" for the port I am working with, which is port 9345.

`Active Internet connections (w/o servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 rancher:48916           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:48898           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:48958           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:56558           172.20.10.14:9345       TIME_WAIT
tcp        0      0 rancher:48988           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:ssh             172.20.10.200:44748     ESTABLISHED
tcp        0      0 rancher:48978           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:56568           172.20.10.14:9345       TIME_WAIT
tcp        0      0 rancher:9345            172.20.10.13:40856      TIME_WAIT
tcp        0      0 rancher:9345            172.20.10.13:40892      TIME_WAIT
tcp        0      0 rancher:56538           172.20.10.14:9345       TIME_WAIT
tcp        0      0 rancher:48924           172.20.10.11:9345       TIME_WAIT
tcp        0      0 rancher:56526           172.20.10.14:9345       TIME_WAIT
udp        0      0 rancher:34185           8.8.8.8:domain          ESTABLISHED
udp        0      0 rancher:41489           8.8.4.4:domain          ESTABLISHED
udp        0      0 rancher:47731           8.8.8.8:domain          ESTABLISHED
udp        0      0 rancher:40760           8.8.8.8:domain          ESTABLISHED `

**Reply by brandond** [reply]
> I attempted to investigate the problem, but what I find strange is that the 'curl' command works intermittently, which is confusing.

Do you perhaps need to add some health checks to your nginx configuration? Is it trying to send connections to servers that aren't ready yet?

**Reply by KarimaFathi** [reply]
the logs in /var/log/niginx are showing : 
`2023/07/27 03:07:40 [error] 1873#1873: *1409 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.10.13, server: 0.0.0.0:9345, upstream: "172.20.10.13:9345", bytes from/to client:0/0, bytes from/to upstream:0/0` 

The thing that I got from the logs is that it's not the Load Balancer (LB) that is refusing the connection, but rather the node server. However, I don't know why. Do you have any clues?

Actually, I have already completed the cluster installation and configuration locally using the same steps without encountering any issues. The main difference in this new environment is that I am behind a proxy server.

**Reply by KarimaFathi** [reply]
I think there is a problem with etcd on the server node, which is preventing other nodes from joining it : 

`Jul 27 21:21:45 HA-Rancher-0 rke2[30163]: time="2023-07-27T21:21:45Z" level=error msg="Failed to get recorded learner progress from etcd: context deadline exceeded"
Jul 27 21:21:48 HA-Rancher-0 rke2[30163]: {"level":"warn","ts":"2023-07-27T21:21:48.201Z","logger":"etcd-client","caller":"v3@v3.5.4-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00096ce00/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing failed to do connect handshake, response: \\\"HTTP/1.1 403 Forbidden\\\\r\\\\nContent-Length: 3454\\\\r\\\\nConnection: keep-alive\\\\r\\\\nContent-Language: en\\\\r\\\\nContent-Type: text/html;charset=utf-8\\\\r\\\\nDate: Thu, 27 Jul 2023 20:41:05 GMT\\\\r\\\\nMime-Version: 1.0\\\\r\\\\nServer: squid/3.5.20\\\\r\\\\nVary: Accept-Language\\\\r\\\\nVia: 1.1 squid (squid/3.5.20)\\\\r\\\\nX-Cache: MISS from squid\\\\r\\\\nX-Cache-Lookup: NONE from squid:3128\\\\r\\\\nX-Squid-Error: ERR_ACCESS_DENIED 0\\\\r\\\\n\\\\r\\\\n<!DOCTYPE html PUBLIC \\\\\\\"-//W3C//DTD HTML 4.01//EN\\\\\\\" \\\\\\\"http://www.w3.org/TR/html4/strict.dtd\\\\\\\">\\\\n<html><head>\\\\n<meta type=\\\\\\\"copyright\\\\\\\" content=\\\\\\\"Copyright (C) 1996-2016 The Squid Software Foundation and contributors\\\\\\\">\\\\n<meta http-equiv=\\\\\\\"Content-Type\\\\\\\" content=\\\\\\\"text/html; charset=utf-8\\\\\\\">\\\\n<title>ERROR: The requested URL could not be retrieved</title>\\\\n<style type=\\\\\\\"text/css\\\\\\\"></style>\\\\n</head><body id=ERR_ACCESS_DENIED>\\\\n<div id=\\\\\\\"titles\\\\\\\">\\\\n<h1>ERROR</h1>\\\\n<h2>The requested URL could not be retrieved</h2>\\\\n</div>\\\\n<hr>\\\\n\\\\n<div id=\\\\\\\"content\\\\\\\">\\\\n<p>The following error was encountered while trying to retrieve the URL: <a href=\\\\\\\"172.20.10.11:2379\\\\\\\">172.20.10.11:2379</a></p>\\\\n\\\\n<blockquote id=\\\\\\\"error\\\\\\\">\\\\n<p><b>Access Denied.</b></p>\\\\n</blockquote>\\\\n\\\\n<p>Access control configuration prevents your request from being allowed at this time. Please contact your service provider if you feel this is incorrect.</p>\\\\n\\\\n<p>Your cache administrator is <a href=\\\\\\\"mailto:root?subject=CacheErrorInfo%20-%20ERR_ACCESS_DENIED&amp;body=CacheHost%3A%20squid%0D%0AErrPage%3A%20ERR_ACCESS_DENIED%0D%0AErr%3A%20%5Bnone%5D%0D%0ATimeStamp%3A%20Thu,%2027%20Jul%202023%2020%3A41%3A05%20GMT%0D%0A%0D%0AClientIP%3A%2010.104.12.25%0D%0A%0D%0AHTTP%20Request%3A%0D%0ACONNECT%20%2F%20HTTP%2F1.1%0AUser-Agent%3A%20grpc-go%2F1.40.0%0D%0AHost%3A%20172.20.10.11%3A2379%0D%0A%0D%0A%0D%0A\\\\\\\">root</a>.</p>\\\\n<br>\\\\n</div>\\\\n\\\\n<hr>\\\\n<div id=\\\\\\\"footer\\\\\\\">\\\\n<p>Generated Thu, 27 Jul 2023 20:41:05 GMT by squid (squid/3.5.20)</p>\\\\n\\\\n</div>\\\\n</body></html>\\\\n\\\"\""}
Jul 27 21:21:48 HA-Rancher-0 rke2[30163]: {"level":"info","ts":"2023-07-27T21:21:48.202Z","logger":"etcd-client","caller":"v3@v3.5.4-k3s1/client.go:210","msg":"Auto sync endpoints failed.","error":"context deadline exceeded"}
`

**Reply by brandond** [reply, marked as answer]
There is not a problem with etcd, the problem is that you still haven't configured your proxy settings correctly, and it is trying to use your proxy to connect to etcd. Note that the `403 Forbidden` response is coming from your `Server: squid/3.5.20` proxy server?

`Jul 27 21:21:45 HA-Rancher-0 rke2[30163]: time="2023-07-27T21:21:45Z" level=error msg="Failed to get recorded learner progress from etcd: context deadline exceeded" Jul 27 21:21:48 HA-Rancher-0 rke2[30163]: {"level":"warn","ts":"2023-07-27T21:21:48.201Z","logger":"etcd-client","caller":"v3@v3.5.4-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00096ce00/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing failed to do connect handshake, response: \\\"HTTP/1.1 403 Forbidden\\\\r\\\\nContent-Length: 3454\\\\r\\\\nConnection: keep-alive\\\\r\\\\nContent-Language: en\\\\r\\\\nContent-Type: text/html;charset=utf-8\\\\r\\\\nDate: Thu, 27 Jul 2023 20:41:05 GMT\\\\r\\\\nMime-Version: 1.0\\\\r\\\\nServer: squid/3.5.20`

**Reply by KarimaFathi** [reply]
Problem resolved, thanks for your help. Now the cluster is ready
